### ISSUE TITLE
Create desktop application

### ISSUE THREAD

**Issue by jfletcher20** [jfletcher20/em-button#21]
The desktop application needs to serve as an all-purpose GUI for interactions between the user and the device, so that non-programmers can reprogram and calibrate and customize its paramteres.
- [x] provide port connection feedback
- [x] provide Bluetooth connection feedback
- [x] provide in-app OLED monitor data display
- [x] settings

Potential extra features:
~~provide real-time digital view of em-button~~
- [x] opening interaction could be done by clicking on a digital model of the em-button
- [x] provide UI for easy and demonstrable STP1.0 protocol request creation

**Comment by jfletcher20** [comment]
A bunch of work has been done on the desktop app without versioning. See commit https://github.com/jfletcher20/em-button/commit/3ee8b1fa63339905e7085bd154cb645d922497c8.

**Comment by jfletcher20** [comment]
- [ ] provide calibration option

**Comment by jfletcher20** [comment]
I'm making a note here: huge success.

There are some minor improvements that can yet be made. Such as
* ensuring the buttons for settings, calibration and to restart the device are disabled when no port is found
* ensuring that if no data loads for the digital OLED monitor, it sends another request
* updating the model that's in the background to the latest model version
* loading actions properly into the actions editing form - this is something I tried to accomplish for some time but didn't manage; for some reason it just won't read the data properly or something

Finally, the UI could probably have been done better, but because of design responsiveness I decided to keep it like this for now. It's only a demo, after all.

But these are all just minor quality-of-life improvements and are not central to the thesis. The desktop application is simply a proof-of-concept on how to interact with the em-button through its API, and as far as I can tell: it's a great success.
The app works better than I expected it to.

![Image](https://github.com/jfletcher20/em-button/assets/100687133/8d17226f-4743-47b4-8720-5388473c0d6c)
